svm/src/account_loader.rs

```diff
@@ -488,12 +438,11 @@ fn load_transaction_accounts<CB: TransactionProcessingCallback>(
             }
 
             let program_index = instruction.program_id_index as usize;
-            let program_usage_pattern = AccountUsagePattern::new(message, program_index);
 
             let Some(LoadedTransactionAccount {
                 account: program_account,
                 ..
-            }) = account_loader.load_account(program_id, program_usage_pattern)
+            }) = account_loader.load_account(program_id, false)
```

Comment:
We should still be checking if the program is writable here

Reply:
this needed to be precise before because it needed to hit the program cache in the same way or else the executable flag could change. now its only purpose is to go to `inspect_account`, and we already triggered all the necessary inspections for this tx above on the initial load. its basically the same as if we directly got the accounts from the `accounts` vec we built

Reply:
Ah right, thanks for the explanation!